pkg/util/test/util.go

```diff
@@ -845,6 +850,51 @@ func ValidateStorageCluster(
 	return nil
 }
 
+func validatePxPodsAnnotations(cluster *corev1.StorageCluster, timeout, interval time.Duration) error {
+	logrus.Debug("Validate PX pods..")
+	t := func() (interface{}, bool, error) {
+		// Get list of PX pods
+		pods, err := coreops.Instance().GetPodsByOwner(cluster.UID, cluster.Namespace)
+		if err != nil || pods == nil {
+			return "", true, fmt.Errorf("failed to get pods for StorageCluster [%s/%s]. Err: %v",
+				cluster.Namespace, cluster.Name, err)
+		}
+
+		for _, pod := range pods {
+			storageNode, err := operatorops.Instance().GetStorageNode(pod.Spec.NodeName, pod.Namespace)
+			if err != nil {
+				return nil, true, fmt.Errorf("failed to get StorageNode [%s] for PX pod [%s], Err: %v", pod.Spec.NodeName, pod.Name, err)
+			}
+
+			safeToEvictAnnotationName := "cluster-autoscaler.kubernetes.io/safe-to-evict"
+			safeToEvictValue, exists := pod.Annotations[safeToEvictAnnotationName]
+			if !exists {
+				return nil, true, fmt.Errorf("px pod [%s] is missing expected annnotation [%s]", pod.Name, safeToEvictAnnotationName)
+			}
+			safeToEvict, err := strconv.ParseBool(safeToEvictValue)
+			if err != nil {
+				return nil, true, fmt.Errorf("failed to convert [%v] to bool, Err: %v", safeToEvictValue, err)
+			}
+
+			// This should validate the annotation cluster-autoscaler.kubernetes.io/safe-to-evict value,
+			// when node is Storage, annotation value should be set to false,
+			// when node is Storageless, annotation value should be set to true
+			if storageNode.Status.NodeAttributes.Storage != nil && *storageNode.Status.NodeAttributes.Storage != !safeToEvict {
+				return nil, true, fmt.Errorf("failed to validate PX pod [%s] annotation [%s] value [%v], storageNode storage value [%v]", pod.Name, safeToEvictAnnotationName, *storageNode.Status.NodeAttributes.Storage, safeToEvict)
```

Comment:
The last two arguments seem interchanged.

Reply:
thx, fixed